### PR TITLE
Implement readiness probe for NATS publisher based on connection status of NATS server(s)

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Implement readiness probe for the NATS publisher proxy based on the connection status of the NATS server(s).

**Related issue(s)**
- https://github.com/kyma-project/kyma/issues/13029

### CODE REVIEW

File: components/event-publisher-proxy/pkg/sender/nats.go [hunk 1]
```diff
@@ -18,48 +18,45 @@ type GenericSender interface {
 	Send(context.Context, *cev2event.Event) (int, error)
 }
 
-// BebMessageSender is responsible for sending messages over HTTP.
+// NatsMessageSender is responsible for sending messages over HTTP.
```
Comment (k15r) [inline components/event-publisher-proxy/pkg/sender/nats.go:21]: 😄

**Comment by k15r** [comment]
the error message sent to the user in case the connection to nats is not working could be improved:
```
Kyma application API responded with status 500 { "status": 500, "message": "connection status not connected: RECONNECTING" }```

**Review by FriedrichWilken — CHANGES_REQUESTED** [review]
there is an issue when port forwarding to epp, while having not nats backend up: when sending an event it gets a 2xx status, while it should be 4xx or 5xx.

**Comment by marcobebway** [comment]
> there is an issue when port forwarding to epp, while having not nats backend up: when sending an event it gets a 2xx status, while it should be 4xx or 5xx.

Thanks for reporting this. I pushed a [commit](https://github.com/kyma-project/kyma/pull/13120/commits/1fc7ebeaaaa6b346bf172231bcb00691accbe436) for the fix.

**Comment by marcobebway** [comment]
> there is an issue when port forwarding to epp, while having not nats backend up: when sending an event it gets a 2xx status, while it should be 4xx or 5xx.

Done.

**Review by FriedrichWilken — APPROVED** [review]

**Pull request merged**